runtime/hydradx/src/benchmarking/route_executor.rs

```diff
@@ -16,232 +16,159 @@
-		deposit,
-		Some(asset_id),
+	LBP::update_pool_data(
+		RawOrigin::Signed(caller).into(),
+		pool_id,
 		None,
-	)
-}
-
-//NOTE: This is necessary for oracle to provide price.
-fn do_lrna_hdx_trade<T: pallet_omnipool::Config>() -> DispatchResult
-where
-	<T as pallet_omnipool::Config>::Currency: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let trader = create_funded_account::<T>("tmp_trader", 0, 100 * ONE, HDX.into());
-
-	fund::<T>(trader.clone(), LRNA.into(), 100 * ONE)?;
-
-	OmnipoolPallet::<T>::sell(RawOrigin::Signed(trader).into(), LRNA.into(), HDX.into(), ONE, 0)
-}
-
-//NOTE: This is necessary for oracle to provide price.
-fn do_lrna_dai_trade<T: pallet_omnipool::Config>() -> DispatchResult
-where
-	<T as pallet_omnipool::Config>::Currency: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let trader = create_funded_account::<T>("tmp_trader", 0, 100 * ONE, DAI.into());
-
-	fund::<T>(trader.clone(), LRNA.into(), 100 * ONE)?;
-
-	OmnipoolPallet::<T>::sell(RawOrigin::Signed(trader).into(), LRNA.into(), DAI.into(), ONE, 0)
-}
-
-fn fund<T: pallet_omnipool::Config>(
-	to: T::AccountId,
-	currency: <T as pallet_omnipool::Config>::AssetId,
-	amount: Balance,
-) -> DispatchResult {
-	CurrencyOf::<T>::deposit(currency, &to, amount)
-}
-
-use frame_support::assert_ok;
-use frame_support::traits::Hooks;
-use hydradx_traits::router::PoolType;
-use pallet_route_executor::Trade;
-use sp_runtime::{DispatchError, DispatchResult, FixedU128, Permill};
-use sp_std::vec;
-
-const SEED: u32 = 1;
-pub const UNITS: Balance = 100_000_000_000;
-
-fn create_funded_account<T: pallet_omnipool::Config>(
-	name: &'static str,
-	index: u32,
-	amount: Balance,
-	currency: <T as pallet_omnipool::Config>::AssetId,
-) -> T::AccountId
-where
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	let caller: T::AccountId = account(name, index, SEED);
-
-	fund::<T>(caller.clone(), currency, amount).unwrap();
-
-	caller
-}
-
-fn set_period<T: pallet_omnipool::Config>(to: u32)
-where
-	T: pallet_ema_oracle::Config,
-	CurrencyOf<T>: MultiCurrencyExtended<T::AccountId, Amount = i128>,
-	<T as pallet_omnipool::Config>::AssetId: From<u32>,
-{
-	while System::<T>::block_number() < to.into() {
-		let b = System::<T>::block_number();
-
-		System::<T>::on_finalize(b);
-		pallet_ema_oracle::Pallet::<T>::on_finalize(b);
-
-		System::<T>::on_initialize(b + 1_u32.into());
-		pallet_ema_oracle::Pallet::<T>::on_initialize(b + 1_u32.into());
+		Some(start),
+		Some(end),
+		None,
+		None,
+		None,
+		None,
+		None,
+	)?;
 
-		System::<T>::set_block_number(b + 1_u32.into());
-	}
+	System::set_block_number(2u32);
+	Ok(())
 }
 
-//TODO: Rebenchmark both buy and sell with dynamic length of route once we have other AMMs in hydra
-
 runtime_benchmarks! {
-	{ Runtime, pallet_route_executor}
+	{Runtime, pallet_route_executor}
 
-	sell {
-		let n in 1..2;
+	sell_in_lbp {
```

Comment:
I guess the only reason for this lbp-specific stuff is to calculate the overhead right?

Would not be better to use some mocked AMMs? The thing is that this sell_in_lbp has just one route included, and maybe would be better to include more amms to properly benchmark the overhead of looping over AMMs and its logic. What do you think?

Reply:
I see! I would not call it totally independent though, for example we have [this balance check](https://github.com/galacticcouncil/HydraDX-node/blob/e2702e877f9ace7bc0bc0d56356920b08d077269/pallets/route-executor/src/lib.rs#L201) after each trade in the route.

But yeah, this is probably a small weight, and if it would really complicates our life, the current solution is perfectly fine if we overestimate a bit.

Reply:
It's independent because there is no code path that depends on a specific AMM. `AMM::calculate_trade` and `AMM::execute_trade` is called for all AMMs. `ensure_that_user_spent_asset_in` is also called every time. The code that belongs to "overhead" doesn't care what AMM we use. It runs the same code for each AMM.

Reply:
Yeah, what I meant is that the loop depends on the number of AMM trades in the loop. Right now in the benchmark the `ensure_that_user_spent_asset_in` is called only once, as the route has one trade. 

But for example for a route with 5 trade, it would be called 5 times. We don't include this in our benchmark.

Reply:
Hmm. You are right. I didn't realize that. But we add the overhead weight for every trade, so the resulting weight is overestimated. I think we can update the benchmarks to calculate them with MaxRouteSize. But it would provide the overhead that is most accurate for trades with MaxRouteSize. The current implementation provides better weights for trades containing just one trade.